### ISSUE TITLE
Shell: Flyoutitems IsVisible set false - what is expected behaviour?

### ISSUE THREAD

**Issue by gfmoore** [dotnet/maui#7769]
### Description

Please refer to https://stackoverflow.com/questions/72463233/net-maui-how-can-i-capture-a-press-click-event-on-a-shell-flyoutitem/72464570#72464570 for history of this behaviour.

Basic upshot.
If no flyout items then error - probably to be expected.

If one or more flyoutitems and IsVisible is set false on all, then also results in app going into break mode
**System.InvalidOperationException:** 'Active Shell Item not set. Have you added any Shell Items to your Shell?'

My expectation would be that there are flyout items and setting the IsVisible property to false should not affect behaviour, but just result in a blank menu

Note that if I have say three items and only one IsVisible set true then fine, if I manually change the IsVisible to false under hot reload the app goes into break mode with the additional error of 
**System.ArgumentOutOfRangeException:** 'Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')'

### Steps to Reproduce

Create a new Maui app
Set the Shell IsVisible flag to "False" on the shell item

Add another two Shell (Flyoutitems) and play around with the IsVisible flag.

### Version with bug

6.0 Release Candidate 3

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android emulator

### Did you find any workaround?

I have a need to change dynamically the menu items on the flyout menu.

In a use case I need to change all flyoutitems and replace them. I am not sure if there are any timing issues with set items false, then others true, or whether just making sure at least one item is visible and changing others then changing the visible on to false,  will work under all conditions

### Relevant log output

```shell
[.shellitemstes] Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (greylist, reflection, allowed)
**System.InvalidOperationException:** 'Active Shell Item not set. Have you added any Shell Items to your Shell?'

and when changing True to False in xaml

Loaded assembly: /data/data/com.companyname.shellitemstest/files/.__override__/System.ComponentModel.Primitives.dll [External]
**System.ArgumentOutOfRangeException:** 'Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')'

Also seems able to kill the android emulator?
```

**Comment by jfversluis** [comment]
I see you've listed "6.0 Release Candidate 3" as a version, is this correct? If so, is there any reason you're not using the GA version?

As for the rest, going by the Docs this should just work: https://docs.microsoft.com/dotnet/maui/fundamentals/shell/flyout#flyoutitem-visibility

**Comment by gfmoore** [comment]
:) That's a good question - I'll update immediately.

Yes, thankyou. I eventually managed to see that little snippet and now it all works well.

**Comment by gfmoore** [comment]
er, struggling to find the instructions on how to upgrade to GA? Honestly search seems so poor nowadays. :(

I've opened the Visual Studio Installer, but it doesn't indicate that .net Maui needs upgrading?
I'm on Visual Studio Community 2022 Preview 17.3.0 Preview 1.1
When I hit modify the .Net Maui platform App UI development is there but to version number etc. But I clicked on it anyway. nd let it do stuff. How do I find out what version of Maui I'm on?

Sorry I guess this should be elsewhere, but you did open a can of worms - :()
